src/NHibernate/Util/EnumerableExtensions.cs

```diff
@@ -6,6 +6,16 @@ namespace NHibernate.Util
 {
 	public static class EnumerableExtensions
 	{
+		public static HashSet<T> ToHashSet<T>(this IEnumerable<T> collection)
```

Comment:
It is likely that others projects define such an extension too. This could cause conflicts for users. Maybe should it be set as `internal` instead of `public`. @hazzik, what do think of this?